core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -202,139 +206,148 @@ private boolean unsubscribe(String queryName,
-                               interceptedQuery.getResponseType(),
-                               interceptedQuery.getUpdateResponseType()));
-            }
-            Iterator<QuerySubscription> subsIterator = subs.iterator();
-            boolean invocationSuccess = false;
-            while (!invocationSuccess && subsIterator.hasNext()) {
-                try {
-                    QuerySubscription subscription = subsIterator.next();
-                    DefaultUnitOfWork<QueryMessage<Q, I>> uow = DefaultUnitOfWork.startAndGet(interceptedQuery);
-                    if (subscription instanceof SubscribableQuerySubscription) {
-                        if (interceptedQuery.getUpdateResponseType()
-                                            .matches(((SubscribableQuerySubscription) subscription).getUpdateType())) {
-
-                            registration = invokeSubscriptionQueryHandler(uow,
-                                                                          ((SubscribableQuerySubscription) subscription)
-                                                                                  .getSubscriptionQueryHandler(),
-                                                                          updateHandler);
-                            invocationSuccess = true;
-                        }
-                    } else {
-                        invokeRegularQueryHandler(uow, subscription.getQueryHandler(), updateHandler);
-                        invocationSuccess = true;
-                    }
-                } catch (NoHandlerForQueryException e) {
-                    // Ignore this Query Handler, as we may have another one which is suitable
-                }
-            }
-            if (!invocationSuccess) {
-                throw new NoHandlerForQueryException(
-                        format("No suitable handler was found for %s with response type %s and update type %s",
-                               interceptedQuery.getQueryName(),
-                               interceptedQuery.getResponseType(),
-                               interceptedQuery.getUpdateResponseType()));
-            }
-            monitorCallback.reportSuccess();
-        } catch (Exception e) {
-            monitorCallback.reportFailure(e);
-            updateHandler.onError(e);
+    public <Q, I, U> SubscriptionQueryResult<QueryResponseMessage<I>, SubscriptionQueryUpdateMessage<U>> subscriptionQuery(
+            SubscriptionQueryMessage<Q, I, U> query,
+            SubscriptionQueryBackpressure backpressure,
+            int updateBufferSize) {
+        boolean alreadyExists = updateHandlers.keySet()
+                                              .stream()
+                                              .anyMatch(m -> m.getIdentifier().equals(query.getIdentifier()));
+        if(alreadyExists) {
+            throw new IllegalArgumentException("There is already a subscription with the same message identifier");
```

Comment:
I'd adjust the message to `There is already a subscription with the given message identifier`, as 'already' and 'same' suggest 'the same' (pun intended).